### PR TITLE
eve::algo::search

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
Still needs more testing and docs.

I think that the tail handling might not be awesome (both for `for_each_selected` and search).

But trashing can start already I think.

### CODE REVIEW

File: include/eve/module/algo/algo/search.hpp [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    wide_value_type_t<I2> first_wide;
+    eve::keep_first       first_wide_ignore; // TODO: use a logical: #1940
+
+    std::ptrdiff_t  long_tail_offset;
+    unaligned_t<I2> long_tail_start;
+    std::ptrdiff_t  long_tail_n;
+
+    template<typename S2>
+    needle_checker(I2 f, S2 l, P _p) : p(_p)
+                                     , first_wide_ignore(iterator_cardinal_v<I2>)
```
[the remark this answers is not in the document]
Comment (jfalcou) [reply include/eve/module/algo/algo/search.hpp:40]: would you want it to be keep_first(0) or somethign else ?

File: include/eve/module/algo/algo/search.hpp [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+//!
+//!  @brief SIMD version of std::search (subsequence in a sequence).
+//!
+//!   **Defined in Header**
+//!
+//!   @code
+//!   #include <eve/module/algo.hpp>
+//!   @endcode
+//!
+//!   Some ideas taken from previous work:
```
Comment (jfalcou) [inline include/eve/module/algo/algo/search.hpp:237]: a bit more doc could be nice. At least repsing the std like prototype and havign a doc/ example maybe ?

Comment (DenisYaroshevskiy) [reply include/eve/module/algo/algo/search.hpp:237]: I will ofc, not finished, sorry

File: include/eve/module/algo/algo/search.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (DenisYaroshevskiy) [reply include/eve/module/algo/algo/search.hpp:40]: Just garbage probably

**Review by jfalcou — APPROVED** [review]

**Pull request merged**